alerting/src/main/kotlin/com/amazon/opendistroforelasticsearch/alerting/MonitorRunner.kt

```diff
@@ -292,7 +301,25 @@ class MonitorRunner(
                         val searchResponse: SearchResponse = client.suspendUntil { client.search(searchRequest, it) }
                         results += searchResponse.convertToMap()
                     }
-                    else -> {
+                    is HttpInput -> {
+                        val response: HttpResponse = httpClient.client.suspendUntil {
```

Comment:
There are a few problems here.  Firstly the read of the content is still blocking as it uses an input stream which blocks waiting for data. Apache Async HttpClient provides APIs for reading the response asynchronously (streaming). See the sample code here: https://hc.apache.org/httpcomponents-asyncclient-dev/examples.html . 

Alternatively to continue to use blocking I/O you need to run this in the I/O dispatcher `Dispatchers.IO`.

Reply:
Decided to change to `results += withContext(Dispatchers.IO) {
                            response.toMap()
                        }`